src/test_application/OSPMagnum.cpp

```diff
@@ -160,76 +162,165 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
 void testapp::config_controls(OSPMagnum& rOspApp)
 {
     // Configure Controls
+    //Load toml
+    auto data = toml::parse("settings.toml");
+    try 
+    {
+        osp::UserInputHandler& rUserInput = rOspApp.get_input_handler();
+        for (const auto& [k, v] : data.as_table())
+        {
+            std::cout << "Parsing " << k << std::endl;
+            std::string primary = toml::find(v, "primary").as_string();
```

Comment:
If you're not going to modify, or move, the `std::string primary` then bind the temporary returned by toml::find to a const& variable, like this:
```
std::string const& primary = toml::find();.
```
The reason to get into the habit of doing this is that you don't typically know (sometimes you do, most times you dont) whether the function you are calling is returning a std::string by *value*, or is instead returning a const& to a std::string that lives somewhere else.

So if you aren't intending to modify / move the string, you don't want to make a copy of the string if the return type is const&, you just want the const&.

If the function returns by value, then the temporary value that is returned can either be stored as a real variable (what you're doing now) or it can be stored as a "temporary with lifetime extension", and then your const& points to that temporary, which is what I'm recommending you do.

Reply:
In the generic sense, there are a few catagories of how to accept return values. But I'll give you the very simplified version.

1) You don't modify or move from the variable. It's "const" as far as the current code cares. Accpet the variable as const&.
2) You *do* modify or move the variable. Accept the variable by value, without const&.